spec/github_pages_health_check/domain_spec.rb

```diff
@@ -151,6 +154,16 @@
       expect(subject.should_be_a_record?).to be_falsy
       expect(subject).to be_a_invalid_a_record
     end
+
+    context "AAAA records" do
+      let(:ip6) { "::1" }
+      before(:each) { allow(subject).to receive(:dns) { [a_packet, aaaa_packet] } }
+      it "knows that AAAA records are not allowed" do
+        expect(subject).to be_an_a_record
+        expect(subject).to be_a_valid_domain
+        expect(subject.invalid_aaaa_record?).to be_truthy
+      end
```

Comment:
Can we also add a test to ensure it raises the correct error?

```ruby
it "raises InvalidAAAARecordError" do 
  expect(-> { subject.check! }).to raise_error(Errors::InvalidAAARecordError)
end
```

This ensures that the checking will surface this error in the right order in the `check!` method.